app/crm/connectivity/api.py

```
@@ -0,0 +1,200 @@
+    template_id: str,
+    req: SubmitTemplateRequest,
+    current_user: UserInfo = Depends(get_current_user_with_rbac),
+) -> TemplateRead:
+    _require_merchant_access(current_user, req.merchant_id, "submit template")
+    set_log_context(component="crm.connectivity.templates", merchant_id=req.merchant_id)
+    try:
+        return await contracts.submit_template(
+            req.merchant_id, template_id, req.category
+        )
+    except (TemplateError, WhatsappProviderError) as e:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))
```

Ruff (0.16.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)